### DISCUSSION TITLE
Document 'shared' interfaces....Cisco 4948 ports45-48

### DISCUSSION THREAD

**Discussion by TheDirkMelvin** [netbox-community/netbox#8412] [Q&A] [answered]
We have some old Cisco 4948 switches that have 'shared' or 'alternate' ports for G1/45-G1/48.
Ports 45-48 have the regular RJ45/Copper ports....but also Ports 45-48 as SFP ports...in the config you don't differentiate between the port names, that are all configured as G1/45 - G1/48.

So if we are using G1/45 and G1/48 (RJ45), but G1/46 and G1/47 (Fiber SFP Transceiver), how in Netbox can we indicate that difference?

Apologies if this is already addressed, I search for 'shared port', 'shared interface', etc. but nothing relevant popped up.

Thanks,
Dirk

**Comment by jeremystretch** [comment, marked as answer]
There's a bit [on the wiki](https://github.com/netbox-community/netbox/wiki/Data-Model-Limitations#combo-ports) about dealing with this type of interface. Unfortunately there's no great solution: The conventional approach is to only model the interface that you actually use.

**Reply by TheDirkMelvin** [reply]
Thanks for the SUPER fast reply

**Comment by mtinberg** [comment]
You can change the port type when editing the interface from 1000base-t to 1000base-x-sfp when you change how it's populated.

On some devices you may have Te1/1/1-2 and Gi1/1/1-4 that share SFP ports and show up as distinct interfaces in the config even though they can't all be populated at the same time; in that case you'd probably model all of them, to match the config and have procedure around the fact that Te1/1/1 can't be populated if Gi1/1/1 is or vice versa, kind of the same as you would if they were being configured by hand.


—

Mark Tinberg ***@***.***>

Division of Information Technology-Network Services

University of Wisconsin-Madison


________________________________
From: Dirk Melvin ***@***.***>
Sent: Thursday, January 20, 2022 9:59 AM
To: netbox-community/netbox ***@***.***>
Cc: Subscribed ***@***.***>
Subject: [netbox-community/netbox] Document 'shared' interfaces....Cisco 4948 ports45-48 (Discussion #8412)


We have some old Cisco 4948 switches that have 'shared' or 'alternate' ports for G1/45-G1/48.
Ports 45-48 have the regular RJ45/Copper ports....but also Ports 45-48 as SFP ports...in the config you don't differentiate between the port names, that are all configured as G1/45 - G1/48.

So if we are using G1/45 and G1/48 (RJ45), but G1/46 and G1/47 (Fiber SFP Transceiver), how in Netbox can we indicate that difference?

Apologies if this is already addressed, I search for 'shared port', 'shared interface', etc. but nothing relevant popped up.

Thanks,
Dirk

—
Reply to this email directly, view it on GitHub<https://github.com/netbox-community/netbox/discussions/8412>, or unsubscribe<https://github.com/notifications/unsubscribe-auth/AAS7UM5ARJ5XKYFKKNEER5LUXAWNPANCNFSM5MNEOCBA>.
Triage notifications on the go with GitHub Mobile for iOS<https://apps.apple.com/app/apple-store/id1477376905?ct=notification-email&mt=8&pt=524675> or Android<https://play.google.com/store/apps/details?id=com.github.android&referrer=utm_campaign%3Dnotification-email%26utm_medium%3Demail%26utm_source%3Dgithub>.

**Comment by candlerb** [comment]
My solution is in the Device Type to create both types of interface with different names:

```
1/0/23F
1/0/23T
1/0/24F
1/0/24T
```

Then I mark the ones I'm not using as "disabled".  This correctly models that there are distinct physical ports; the downside is that interface names no longer exactly match the interface names in the config

You *could* create a virtual interface 1/0/24, and set its parent to be 1/0/24F or 1/0/24T as appropriate.
